src/plain_printer.cc

```diff
@@ -72,41 +72,62 @@ std::vector<std::string> champsim::plain_printer::format(CACHE::stats_type stats
   using mshr_merge_value_type = typename decltype(stats.mshr_merge)::value_type;
   using mshr_return_value_type = typename decltype(stats.mshr_return)::value_type;
 
+  std::vector<std::size_t> cpus;
+
+  //build a vector of all existing cpus
+  auto stat_keys = {stats.hits.get_keys(), stats.misses.get_keys(), stats.mshr_merge.get_keys(), stats.mshr_return.get_keys()};
+  for (auto keys : stat_keys)
+    for (auto key : keys)
+      if(std::find_if(cpus.begin(), cpus.end(), [key=key](const auto val) {return key.second == val;}) == cpus.end())
+        cpus.push_back(key.second);
```

Comment:
```suggestion
  for (auto keys : stat_keys) {
    std::transform(std::begin(keys), std::end(keys), std::back_inserter(cpus), [](auto val){ return val.second; });
  }
  std::sort(std::begin(cpus), std::end(cpus));
  auto uniq_end = std::unique(std::begin(cpus), std::end(cpus));
  cpus.erase(uniq_end, std::end(cpus));
```
This avoids searching the vector dozens of times.